examples/custom_image_embedding_search.ipynb

```diff
@@ -0,0 +1,484 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "w9w5JBaUL-lO"
+      },
+      "source": [
+        "# CLIP embeddings to improve multimodal RAG with GPT-4 Vision\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "3CCjcFSiMbvf"
+      },
+      "source": [
+        "Multimodal RAG supercharges the traditional text-based RAG with other modalities to provide further useful context to an LLM to help answer questions. Other modalities may hold information required to answer questions which do not exist within the text or they provide useful grounding for the textual data to improve the LLM's understanding.\n",
+        "\n",
+        "Building on work done in the [clothing matchmaker cookbook](https://cookbook.openai.com/examples/how_to_combine_gpt4v_with_rag_outfit_assistant), instead of using GPT-4V to caption an image and perform RAG on the caption, we instead embed the image directly as use that as the basis of our similarity search. The reason for this is in certain situations, there is information loss in converting an image to text as the text cannot fully capture the richness of the image. Therefore this direct image embedding may improve the most crucial part of RAG, the retrieval.\n",
+        "\n",
+        "An additional benefit of using CLIP-based embeddings is that we are also able to finetune CLIP to an enterprises' proprietary data or to update it with images it has not yet seen.\n",
+        "\n",
+        "The application area we will use to bring this technique to life is performing RAG over user inputted images of technology and searching our enterprise knowledge base for a matching image to provide helpful information for the end user."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "nbt3evfHUJTZ"
+      },
+      "source": [
+        "### Installations and imports\n",
+        "First let's install the relevant packages."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7hgrcVEl0Ma1"
+      },
+      "outputs": [],
+      "source": [
+        "#installations\n",
+        "!pip install clip\n",
+        "!pip install torch\n",
+        "!pip install pillow\n",
+        "!pip install faiss-cpu\n",
+        "!pip install numpy\n",
+        "!pip install git+https://github.com/openai/CLIP.git\n",
+        "!pip install openai"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "GgrlBLTpT0si"
+      },
+      "source": [
+        "Then let's import all the needed packages.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 44,
```

Comment:
nit: Make sure to re-run your file in order so these are sequential and consecutive